components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/X509CredentialImpl.java

```diff
@@ -43,8 +60,184 @@
 public class X509CredentialImpl implements X509Credential {
 
     private PublicKey publicKey = null;
+    private PrivateKey privateKey = null;
     private X509Certificate signingCert = null;
 
+    private static KeyStore superTenantSignKeyStore = null;
+
+    private static Log log = LogFactory.getLog(X509CredentialImpl.class);
+
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_LOCATION = "Security.SAMLSignKeyStore.Location";
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_TYPE = "Security.SAMLSignKeyStore.Type";
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_PASSWORD = "Security.SAMLSignKeyStore.Password";
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_KEY_ALIAS = "Security.SAMLSignKeyStore.KeyAlias";
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_KEY_PASSWORD = "Security.SAMLSignKeyStore.KeyPassword";
+
+    /**
+     * Instantiates X509Credential.
+     * This credential object will hold the private key, public key and the cert for the respective tenant domain.
+     *
+     * @param tenantDomain tenant domain
+     */
+    public X509CredentialImpl(String tenantDomain) throws IdentityException {
+
+        int tenantId;
+        try {
+            tenantId = SAMLSSOUtil.getRealmService().getTenantManager().getTenantId(tenantDomain);
+        } catch (UserStoreException e) {
+            throw new IdentityException("Exception occurred while retrieving Tenant ID from tenant domain " +
+                    tenantDomain, e);
+        }
+
+        KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(tenantId);
+
+        // Get the private key and the cert for the respective tenant domain.
+        if (!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+            initializeCredentialForTenant(tenantDomain, keyStoreManager);
+        } else {
+            if (isSignKeyStoreConfigured()) {
+                initializeCredentialForSuperTenantFromSignKeyStore();
+            } else {
+                initializeCredentialForSuperTenantFromDefaultKeyStore(keyStoreManager);
+            }
+        }
+
+        if (privateKey == null) {
+            throw new IdentityException("Cannot find the private key for tenant " + tenantDomain);
+        }
+
+        if (signingCert == null) {
+            throw new IdentityException("Cannot find the certificate.");
+        }
+
+        publicKey = signingCert.getPublicKey();
+    }
+
+    /**
+     * Set private key and X509Certificate from the default KeyStore.
+     *
+     * @param keyStoreManager keyStore Manager.
+     * @throws IdentityException Error in retrieving private key and certificate.
+     */
+    private void initializeCredentialForSuperTenantFromDefaultKeyStore(KeyStoreManager keyStoreManager)
```

Comment:
What does default key store means here ? I am just trying to shorten the method name.. We might be able to use init instead of initialize.